src/lib/paginator/paginator.spec.ts

```diff
@@ -0,0 +1,287 @@
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('0 of 0');
+
+      // View third page of list of 10, each page contains 5 items.
+      component.listLength = 10;
+      component.pageLength = 5;
+      component.currentPageIndex = 2;
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('11 - 15 of 10');
+
+      // View third page of list of -5, each page contains 5 items.
+      component.listLength = -5;
+      component.pageLength = 5;
+      component.currentPageIndex = 2;
+      fixture.detectChanges();
+      expect(rangeElement.innerText).toBe('11 - 15 of 0');
+    });
+
+    it('should show right aria-labels for select and buttons', () => {
+      expect(fixture.nativeElement.querySelector('.mat-select'))
+          .toHaveAriaLabel('Items per page:');
+      expect(fixture.nativeElement.querySelector('.mat-paginator-navigation-previous'))
+          .toHaveAriaLabel('Previous page');
+      expect(fixture.nativeElement.querySelector('.mat-paginator-navigation-next'))
+          .toHaveAriaLabel('Next page');
+    });
+  });
+
+  describe('when navigating with the navigation buttons', () => {
+    it('should be able to go to the next page', () => {
+      expect(paginator.currentPageIndex).toBe(0);
+
+      component.clickNextButton();
+
+      expect(paginator.currentPageIndex).toBe(1);
+      expect(component.latestPageChangeEvent.currentPageIndex).toBe(1);
+    });
+
+    it('should be able to go to the previous page', () => {
+      paginator.currentPageIndex = 1;
+      fixture.detectChanges();
+      expect(paginator.currentPageIndex).toBe(1);
+
+      component.clickPreviousButton();
+
+      expect(paginator.currentPageIndex).toBe(0);
+      expect(component.latestPageChangeEvent.currentPageIndex).toBe(0);
+    });
+
+    it('should disable navigating to the next page if at first page', () => {
+      component.goToLastPage();
+      fixture.detectChanges();
+      expect(paginator.currentPageIndex).toBe(10);
+      expect(paginator.canNavigateToNextPage()).toBe(false);
+
+      component.latestPageChangeEvent = null;
+      component.clickNextButton();
+
+      expect(component.latestPageChangeEvent).toBe(null);
+      expect(paginator.currentPageIndex).toBe(10);
+    });
+
+    it('should disable navigating to the previous page if at first page', () => {
+      expect(paginator.currentPageIndex).toBe(0);
+      expect(paginator.canNavigateToPreviousPage()).toBe(false);
+
+      component.latestPageChangeEvent = null;
+      component.clickPreviousButton();
+
+      expect(component.latestPageChangeEvent).toBe(null);
+      expect(paginator.currentPageIndex).toBe(0);
+    });
+  });
+
+  it('should fail if page length and page length options are not provided', () => {
+    expect(() => {
+      TestBed.createComponent(MdPaginatorWithoutPageLengthOrOptionsApp).detectChanges();
+    }).toThrowError('No page length options available for the MdPaginator');
```

Comment:
The error and the test will be out of sync if we decide to change the wording. Perhaps move it into a function (see `input-container-errors.ts`)?